### PR TITLE
internal/bgp/native: use SetsockoptTCPMD5Sig from golang.org/x/sys/unix

### PR DESCRIPTION
**Pull request by tklauser**
Use the TCPMD5Sig type and the corresponding SetsockoptTCPMD5Sig func added upstream in golang.org/x/sys v0.6.0

### CODE REVIEW

File: internal/bgp/native/native.go [hunk 1]
```diff
@@ -665,20 +658,20 @@ func dialMD5(ctx context.Context, addr string, srcAddr net.IP, password string)
 	}
 }
 
-func buildTCPMD5Sig(addr net.IP, key string) tcpmd5sig {
```
Comment (fedepaol) [inline internal/bgp/native/native.go:668]: we also have the struct definition we can get rid of, I guess

Comment (tklauser) [reply internal/bgp/native/native.go:668]: Thanks, removed the struct definition as well.

**Comment by fedepaol** [comment]
LGTM, and sorry for the delay!

**Comment by tklauser** [comment]
@fedepaol I might need some help to figure out the [e2e-use-operator test failure](https://github.com/metallb/metallb/actions/runs/4418139877/jobs/7745093780). I'm not sure whether it is related to this PR or an existing flake/known failure.

**Comment by fedepaol** [comment]
> @fedepaol I might need some help to figure out the [e2e-use-operator test failure](https://github.com/metallb/metallb/actions/runs/4418139877/jobs/7745093780). I'm not sure whether it is related to this PR or an existing flake/known failure.

Ignore the failure, it's a chicken - egg issue related to the fact that we upgraded the frr version on metallb but not yet on the operator. I am merging the prs in a fifo manner, will merge this when I get to it

**Pull request merged**